packages/lwc/types.d.ts

```diff
@@ -195,33 +195,34 @@ declare module 'lwc' {
      */
     export function wire(getType: (config?: any) => any, config?: any): PropertyDecorator;
 
-    type WireConfigValue = Record<string, any>;
-    type ContextValue = Record<string, any>;
-
-    interface WireAdapter {
-        update(config: WireConfigValue, context?: ContextValue): void;
-        connect(): void;
-        disconnect(): void;
+    interface WireAdapterConstructor<Value, Config, Context> {
```

Comment:
i do like this generics use, i have a few questions and a request:

- should this types match the ones exported by https://github.com/salesforce/lwc/blob/master/packages/%40lwc/engine-core/src/framework/main.ts#L45-L52  ?
- do you think is worth to replace the existing types in `@lwc/engine-core` with these generic ones?

can you create some examples of code using this new types?